[OpenCL] Add OpenCL CPU RT code

cdai2 opened the pull request:
It includes the OpenCL CPU RT, Transform passes to support SYCL and OCL and LIT test cases for them.
.

cdai2 commented:
The compilation support for this code is not included yet. We will submit it in another patch. It only contains the code we want to land.

bader commented on buildbot/configure.py:
```diff
@@ -65,6 +68,15 @@ def do_configure(args):
+
+    if args.bldocl or args.bldfpgaemu:
+        llvm_external_projects += ';opencl-cpu'
+
```
These components should be enabled when args.ci_defaults is true to build them in CI.

bader replied:
@cdai2, please, address this comment.

I also recommend moving "Transform passes to support SYCL and OCL and LIT test cases for them" to a separate patch. It doesn't require changes in build scripts except CMake changes and will make it easier to review.

Please, provide instructions for reviewers how to review changes. GitHub is not able to display the full diff.

cdai2 replied:
I tried to enable opencl-cpu build under args.ci_defaults arguments according to my understanding.  Please correct me if it's not true.

bader replied:
> As this PR is to upload these files for the first time. I have no idea to display the diff for it.

I think a major part of the diff is the content of `opencl/` directory. I suggest we keep changes in files outside of this directory in separate commits and you can ask code owners to review specific commits only and ignore the commits with changes in `opencl/` directory. I assume that @intel/ocl-cpu-rt-write team is already reviewed the content of the `opencl/` directory and can approve it. If not, the only idea I have is to pull the source code locally and review it locally.

hewj03 commented:
@cdai2 pass registration code isn't uploaded yet?

cdai2 commented:
> @cdai2 pass registration code isn't uploaded yet?

landed again.

jbrodman commented:
Why are there folders and files called SYCL Transforms if this is an OpenCL compiler?

jbrodman commented:
If you think they're more general than OpenCL and SYCL, I'd suggest "SPMD" as a better name than SYCL then.

cdai2 commented:
> If you think they're more general than OpenCL and SYCL, I'd suggest "SPMD" as a better name than SYCL then.
We think SYCL may be more straightforward than SPMD for these passes.  Our CPU device is now oriented for SYCL.

cdai2 commented:
@bader   Could you please help to remove unnecessary reviewers? I did a rebase and squashed my two commits. Then a lot of commits are contained and introduced these reviewers.

bader commented:
> @bader Could you please help to remove unnecessary reviewers? I did a rebase and squashed my two commits. Then a lot of commits are contained and introduced these reviewers.

Do you mean you can't remove them?

jbrodman commented:
Why are there binary .dlls included in this PR?

jbrodman commented:
There are also several external dependencies that have been copied into this source tree. Shouldn't these be dynamically pulled in by the configuration step?

cdai2 commented:
> Why are there binary .dlls included in this PR?

They are generated during compilation for test cases.  We will remove. thanks

cdai2 commented:
> There are also several external dependencies that have been copied into this source tree. Shouldn't these be dynamically pulled in by the configuration step?

Yes, We have plan to pull in them during configuration step.  we will update it. thanks

ValeZAA commented:
> @stdale-intel - Can this be closed?

Was it even ever merged?
